security/tutorials/calico-policy.md

```diff
@@ -0,0 +1,345 @@
+> {: .alert .alert-info}
+
+### 5. Allow DNS egress traffic
+
+Run the following to create a label of `name: kube-system` on the `kube-system` namespace and a `NetworkPolicy` which allows DNS egress traffic
+from any pods in the `advanced-policy-demo` namespace to the `kube-system` namespace.
+
+```bash
+kubectl label namespace kube-system name=kube-system
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-dns-access
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  egress:
+  - action: Allow
+    protocol: UDP
+    destination:
+      namespaceSelector: name == 'kube-system'
+      ports:
+      - 53
+
+EOF
+```
+
+#### Verify access - allowed DNS access
+
+Now egress traffic to DNS will be allowed.
+
+We can see that this is the case by switching over to our "access" pod in the namespace and attempting to lookup nginx and google.com.
+
+```bash
+nslookup nginx
+```
+
+It should return something like the following.
+
+```
+Server:    10.0.0.10
+Address 1: 10.0.0.10 kube-dns.kube-system.svc.cluster.local
+```
+
+{: .no-select-button}
+
+Next, try to look up google.com.
+
+```bash
+nslookup google.com
+```
+
+It should return something like the following.
+
+```
+Name:      google.com
+Address 1: 2607:f8b0:4005:807::200e sfo07s16-in-x0e.1e100.net
+Address 2: 216.58.195.78 sfo07s16-in-f14.1e100.net
+```
+
+{: .no-select-button}
+
+Even though DNS egress traffic is now working, all other egress traffic from all pods in the advanced-policy-demo namespace is still blocked. Therefore the HTTP egress traffic from the `wget` calls will still fail.
+
+### 6. Allow egress traffic to nginx
+
+Run the following to create a `NetworkPolicy` which allows egress traffic from any pods in the `advanced-policy-demo` namespace to pods with labels matching `run: nginx` in the same namespace.
+
+```bash
+calicoctl create -f - <<EOF
+apiVersion: projectcalico.org/v3
+kind: NetworkPolicy
+metadata:
+  name: allow-egress-to-advanced-policy-ns
+  namespace: advanced-policy-demo
+spec:
+  selector: all()
+  egress:
+  - action: Allow
+    destination:
+      selector: run == 'nginx'
+EOF
+```
+
+#### Verify access - allowed egress access to nginx
+
+We can see that this is the case by switching over to our "access" pod in the
+namespace and attempting to access `nginx`.
+
+```bash
+wget -q --timeout=5 nginx -O -
+```
+
+It should return the HTML of the nginx welcome page.
+
+```
+<!DOCTYPE html>
+<html>
+<head>
+<title>Welcome to nginx!</title>...
+```
+
+{: .no-select-button}
+
+Next, try to retrieve the home page of google.com.
+
+```bash
+wget -q --timeout=5 google.com -O -
+```
+
+It should return:
+
+```
+wget: download timed out
+```
+
+{: .no-select-button}
+
+Access to `google.com` times out because it can resolve DNS but has no egress access to anything other than pods with labels matching `run: nginx` in the `advanced-policy-demo` namespace.
```

Comment:
Access to `google.com` times out because it can resolve DNS, but

Reply:
Fixed!